File: tests/python/anthropic/test_openinference.py
"""Conformance test: OpenInference (Arize) Anthropic instrumentation."""

from common import run


def instrument():
    from openinference.instrumentation.anthropic import AnthropicInstrumentor
    from openinference.instrumentation.config import TraceConfig

    AnthropicInstrumentor().instrument(config=TraceConfig(enable_genai_semconv=True))


if __name__ == "__main__":
    run("OpenInference: Anthropic Conformance Test", instrument)

File: tests/js/anthropic/test_openinference.ts
/**
 * Conformance test: OpenInference (Arize) JS Anthropic instrumentation.
 */

import { AnthropicInstrumentation } from "@arizeai/openinference-instrumentation-anthropic";
import { run } from "./common";

function instrument(anthropicModule: any) {
  const instrumentation = new AnthropicInstrumentation();
  instrumentation.manuallyInstrument(anthropicModule);
}

run("OpenInference JS: Anthropic Conformance Test", instrument).catch((e) => { console.error(e); process.exit(1); });
